packages/contracts-bedrock/test/L2/GasPriceOracle.t.sol

```diff
@@ -468,19 +492,71 @@ contract GasPriceOracleJovian_Test is GasPriceOracle_Test {
-        assertEq(isthmusFee, 10 * operatorFeeScalar / 1e6 + operatorFeeConstant);
+        assertEq(
+            isthmusFee,
+            uint256(10) * operatorFeeScalar / 1e6 + operatorFeeConstant,
+            "Isthmus operator fee incorrect with 10 gas used"
+        );
+
+        // Use maximum values permitted by data types for scalars.
+        // Use maximum value for gasUsed according to client implementations.
+        // Assert that the fee is as expected (no overflow).
+        _setOperatorFeeParams(MAX_UINT32, MAX_UINT64);
+        isthmusFee = gasPriceOracle.getOperatorFee(MAX_UINT64);
+        assertEq(
+            isthmusFee,
+            uint256(MAX_UINT64) * MAX_UINT32 / 1e6 + MAX_UINT64,
+            "Isthmus operator fee incorrect with max uint64 gas used"
+        );
+
+        // Show that the math saturates if the maximum
+        // value for gasUsed (according to data type) is used.
+        _setOperatorFeeParams(1e6, 1);
+        uint256 saturatedIsthmusFee = gasPriceOracle.getOperatorFee(MAX_UINT256);
+        assertEq(
+            saturatedIsthmusFee,
+            115792089237316195423570985008687907853269984665640564039457584007913130,
+            "Incorrect value for fee under Isthmus (saturating arithmetic triggered)"
+        );
 
         // Activate Jovian
         _activateJovian();
 
-        // Check Jovian formula (multiply by 100)
+        // Check Jovian formula with a low gasUsed value (multiply by 100)
+        _setOperatorFeeParams(operatorFeeScalar, operatorFeeConstant);
         uint256 jovianFee = gasPriceOracle.getOperatorFee(10);
-        assertEq(jovianFee, 10 * operatorFeeScalar * 100 + operatorFeeConstant);
+        assertEq(
+            jovianFee,
+            uint256(10) * operatorFeeScalar * 100 + operatorFeeConstant,
+            "Jovian operator fee incorrect with 10 gas used"
+        );
+
+        // Use maximum values permitted by data types for scalars.
+        // Use maximum value for gasUsed according to client implementations.
+        // Assert that the fee is as expected (no overflow).
+        _setOperatorFeeParams(MAX_UINT32, MAX_UINT64);
+        jovianFee = gasPriceOracle.getOperatorFee(MAX_UINT64);
+        assertEq(
+            jovianFee,
+            uint256(MAX_UINT64) * MAX_UINT32 * 100 + MAX_UINT64,
+            "Jovian operator fee incorrect with max uint64 gas used"
+        );
+
+        // Show that a revert is possible if the maximum
+        // value for gasUsed (according to data type) is used.
+        _setOperatorFeeParams(1, 1);
+        vm.expectRevert(stdError.arithmeticError);
+        gasPriceOracle.getOperatorFee(MAX_UINT256);
 
         // Verify the fee increased significantly
-        assertGt(jovianFee, isthmusFee);
+        assertGt(jovianFee, isthmusFee, "Jovian formula fee should be greater than Isthmus formula fee");
     }
 }
+
+library stdError {
+    bytes public constant arithmeticError = abi.encodeWithSignature("Panic(uint256)", 0x11);
+}
```

Comment:
you can just import this library from forge-std https://github.com/foundry-rs/forge-std/blob/b8f065fda83b8cd94a6b2fec8fcd911dc3b444fd/src/StdError.sol